src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+        assert (
+            templates.num_channels == recording.get_num_channels()
+        ), "templates and recording should have the same number of channels"
+        num_units = templates.num_units
+        nbefore = templates.nbefore
+        nafter = templates.nafter
+        unit_locations = compute_monopolar_triangulation(templates)
+
+        channel_locations_rel = channel_locations - channel_locations[0]
+        templates_locations = templates.get_channel_locations()
+        templates_locations_rel = templates_locations - templates_locations[0]
+
+        if not np.allclose(channel_locations_rel, templates_locations_rel):
+            warnings.warn("Channel locations are different between recording and templates. Interpolating templates.")
+            templates_array = np.zeros(templates.templates_array.shape, dtype=dtype)
+            for i in range(len(templates_array)):
+                src_template = templates.templates_array[i][np.newaxis, :, :]
+                templates_array[i] = interpolate_templates(src_template, templates_locations_rel, channel_locations_rel)
+        else:
+            templates_array = templates.templates_array
+
+        # manage scaling of templates
+        templates_ = templates
+        if recording.has_scaleable_traces():
+            if templates_in_uV:
+                templates_array = (templates_array - recording.get_channel_offsets()) / recording.get_channel_gains()
+                # make a copy of the templates and reset templates_array (might have scaled templates)
+                templates_ = templates.select_units(templates.unit_ids)
+                templates_.templates_array = templates_array
+
+    if sorting is None:
+        generate_sorting_kwargs = generate_sorting_kwargs.copy()
+        generate_sorting_kwargs["durations"] = durations
+        generate_sorting_kwargs["sampling_frequency"] = sampling_frequency
+        generate_sorting_kwargs["seed"] = seed
+        generate_sorting_kwargs["num_units"] = num_units
+        sorting = generate_sorting(**generate_sorting_kwargs)
+    else:
+        num_units = sorting.get_num_units()
+        assert sorting.sampling_frequency == sampling_frequency
+
+    num_spikes = sorting.to_spike_vector().size
+    sorting.set_property("gt_unit_locations", unit_locations)
+
+    assert (nbefore + nafter) == templates_array.shape[
+        1
+    ], "templates and ms_before, ms_after should have the same length"
+
+    if templates_array.ndim == 3:
+        upsample_vector = None
+    else:
+        if upsample_vector is None:
+            upsample_factor = templates_array.shape[3]
+            upsample_vector = rng.integers(0, upsample_factor, size=num_spikes)
+
+    if amplitude_std is not None:
+        amplitude_factor = rng.normal(loc=1, scale=amplitude_std, size=num_spikes)
+    else:
+        amplitude_factor = None
+
+    if motion_info is not None:
```

Comment:
I think we should merge the Motion object first here because it will force to many plainfull changes after.